packages/gatsby/src/internal-plugins/partytown/gatsby-ssr.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react"
+import type { GatsbySSR } from "gatsby"
+import { Partytown } from "@builder.io/partytown/react"
+import { PartytownContext } from "gatsby-script"
+import type { PartytownProps } from "@builder.io/partytown/react"
+
+const collectedScripts: Record<string, Array<PartytownProps>> = {}
```

Comment:
I would change this to `collectedScriptForwards` as we don't need collecting anything else other than `forward`? Also would use `Map` instead of object as it's generally faster

Reply:
Ah, we need scripts only to know if there any on the page. Then maybe this is fine as is (I would still say using `Map` would be advisable).

Other thing is that we never remove from `collectedScripts` after using it in `onRenderBody` which make it a memory leak - we should remove entries from this after they are handled